src/fast-path.js

```diff
@@ -405,15 +405,49 @@ FastPath.prototype.needsParens = function() {
         parent.type === "IntersectionTypeAnnotation"
       );
 
-    case "NumericLiteral":
-    case "Literal":
+    case "Literal": {
+      const isDirectiveAllowed = (node, nodes) => {
```

Comment:
I'm not sure performance wise if creating the function inline is a good idea, but I followed another the codestyle from above (`BinaryExpression`)